### ISSUE TITLE
Add Last Tested Yaml To Lessons

### ISSUE THREAD

**Issue by ZoeLeBlanc** [programminghistorian/jekyll#1938]
Originating from #1917 discussion with @acrymble and @svmelton about how to help bug testing process and also handling reporting of bugs.

Idea is to update lesson yaml to include last tested date field and tester name field so that we know how recently the lesson was reviewed for its technical content. (Might also need to think if we want to keep this field as a type of internal history... so also include previous testers and testing dates).

This is likely something that we should discuss as @programminghistorian/technical-team since we might want to discuss if we want to include this as a part of a more robust progress for assessing PH infrastructure documentation

**Comment by ZoeLeBlanc** [comment]
From our #1941 meeting, @rivaquiroga suggested having translators also include that they tested the lessons so we'll need to update the guidelines as well.

**Comment by jenniferisasi** [comment]
For meeting:
My small concern is that our header is already packed, specially when a lesson is a translation? (originals are ok)

**Comment by ZoeLeBlanc** [comment]
Think that's a really good point about being packed. Just to give us an idea this is what it looks like on the latest spanish lesson (I didn't translate the fields though so that might impact space constraints). 
<img width="1438" alt="Screen Shot 2021-03-09 at 8 36 13 PM" src="https://user-images.githubusercontent.com/8355129/110562259-29c5ec80-8117-11eb-988c-fa0072a07839.png">
I just hard coded in those values in snippets and then the lesson, so you would need to do the same to test it on PR #1994

**Comment by jenniferisasi** [comment]
Oh, it doesn't look as bad as I thought. Would be almost the last thing to add in there though. In the lower bar, an option could be to put the information below the metadata fields, so just like the bar above. 

Now... sorry for coming up with another idea/request I cannot implement myself (I think I could try): Can we make it so that when you make the window smaller the information on the metadata fields of the first bar get to the right (like the second bar) so that part gets smaller (less scrolling)? Of course for multiple names we would have to add a conditional to include commas in there. 
<img width="731" alt="Screen Shot 2021-03-09 at 9 37 07 PM" src="https://user-images.githubusercontent.com/11022533/110567484-9ba23400-811f-11eb-8e6c-bccadff6c3cd.png">

**Comment by walshbr** [comment]
You could also hide particular fields if you wanted on smaller windows. Mobile is the obvious extreme, but it'd be worth checking what happens on, say, an iPad, or  a window that is 600px or so.

**Comment by jenniferisasi** [comment]
This one is also an extreme example of 6 people as translators

**Comment by ZoeLeBlanc** [comment]
from #2047 discussions:
Ipad: horizontal for both orientations
full screen: horizontal with columns
mobile: vertical with horizontal rows

**Comment by ZoeLeBlanc** [comment]
@jenniferisasi I'm struggling a bit to remember what each of those orientations means above ☝🏽 

looking at a current lesson this is what iPad looks like vertically (which is already too crowded)
![Screen Shot 2021-03-23 at 7 33 39 PM](https://user-images.githubusercontent.com/8355129/112232342-c6f04d00-8c0e-11eb-8464-ca6f4d2ac897.png)
then iPad horizontal orientation
![Screen Shot 2021-03-23 at 7 33 49 PM](https://user-images.githubusercontent.com/8355129/112232437-eedfb080-8c0e-11eb-919b-e2156e294f9f.png)
And finally mobile vertical
![Screen Shot 2021-03-23 at 7 34 02 PM](https://user-images.githubusercontent.com/8355129/112232453-f606be80-8c0e-11eb-9561-37ecd326004b.png)

do you remember what we meant by horizontal for both orientations for iPad? or vertical with horizontal rows for mobile? Sorry to bug you but I just can't seem to figure it out 😅 

Thanks for your help!!

**Comment by jenniferisasi** [comment]
I think I do, now let's see if I can explain it with words, if not I'll draw it. 

We want to keep horizontal as it is (iPad or computer) because if the screen is wide enough, it's good. 

My suggestion was more for mobile, to make the list in the dark grey are look like the list on the clear grey area. 

Instead of:
```
TRADUCCIÓN EDITADA POR
Jennifer Isasi
Antonio Rojas Castro 
```
make it look like (for vertical, in small screens):
```
**TRADUCCIÓN EDITADA POR** Jennifer Isasi, Antonio Rojas Castro 
```

so that when there is a bunch of people that area is not that long. Makes sense like this? 

Now...for the vertical on an iPad, which is a mid-way in between computer an iPhone... I think it is still good cos the fields of metadata are still like this: metadata | metadata | metadata 

Again, this is a thing for the eye more than anything. If it is too much of a hassle, we can skip it.

**Comment by ZoeLeBlanc** [comment]
Thanks for this clarification! It's super helpful and I still like the idea but I do have some hesitation after looking at a few translated lessons on mobile.

In this screenshot 👇🏽  from a recent translated lesson you can see that if `TRADUCIDO POR` was horizontal it would be impossible to keep it on the screen without having to wrap the names which would essentially give us a less readable list.
<img width="307" alt="Screen Shot 2021-03-24 at 7 49 17 AM" src="https://user-images.githubusercontent.com/8355129/112305809-811a9f80-8c75-11eb-9fac-3c3ac4ea39be.png">

So I'm wondering how strongly you feel about this horizontal layout (given that we have limited horizontal space and unlimited vertical space on mobile)? It might also be worth breaking this out into a separate ticket on lesson metadata layout overall to test different configurations. Let me know what you think and thanks for all your help @jenniferisasi !

**Comment by jenniferisasi** [comment]
No strong feelings about this; if you think it will be messier than now (that is not messy, just long) let's just forget I suggested the edits :)

**Comment by acrymble** [comment]
@ZoeLeBlanc I have a student about to start on testing lessons for us. I don't need the display ready so there's no hurry on closing this. But can you please tell us the YAML format so that we can start adding the metadata correctly as we go?

Just a simple example of use is great.

**Comment by ZoeLeBlanc** [comment]
@acrymble the branch is just waiting for Spanish translation so hopefully we can get that in the next few days.

The update for the lesson yaml will be adding these fields to existing ones:
```
lesson-testers: 
- ADD NAME
- ADD NAME
tested-date: ADD DATE (format YEAR-MONTH-DAY)
```
If your student has feedback on the formatting, would be happy to tweak things and let me know if they have more questions.

**Comment by ZoeLeBlanc** [comment]
Closed with PR #1994
